spanner/cloud-client/backup_sample.py

```diff
@@ -0,0 +1,273 @@
+    operation = backup.create()
+
+    # Wait for backup operation to complete.
+    operation.result()
+
+    # Verify that the backup is ready.
+    backup.reload()
+    assert backup.is_ready() is True
+
+    # Get the name, create time and backup size.
+    backup.reload()
+    print("Backup {} of size {} bytes was created at {}".format(
+        backup.name, backup.size_bytes, backup.create_time))
+# [END spanner_create_backup]
+
+
+# [START spanner_restore_database]
+def restore_database(instance_id, new_database_id, backup_id):
+    """Restores a database from a backup."""
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    # Create a backup on database_id.
+
+    # Start restoring backup to a new database.
+    backup = instance.backup(backup_id)
+    new_database = instance.database(new_database_id)
+    operation = new_database.restore(backup)
+
+    # Wait for restore operation to complete.
+    operation.result()
+
+    # Newly created database has restore information.
+    new_database.reload()
+    restore_info = new_database.restore_info
+    print("Database {} restored to {} from backup {}.".format(
+        restore_info.backup_info.source_database,
+        new_database_id,
+        restore_info.backup_info.backup))
+# [END spanner_restore_database]
+
+
+# [START spanner_cancel_backup]
+def cancel_backup(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+    expire_time = datetime.utcnow() + timedelta(days=30)
+
+    # Create a backup.
+    backup = instance.backup(
+        backup_id, database=database, expire_time=expire_time)
+    operation = backup.create()
+
+    # Cancel backup creation.
+    operation.cancel()
+
+    # Cancel operations are best effort so either it will complete or
+    # be cancelled.
+    while not operation.done():
+        time.sleep(300)  # 5 mins
+
+    # Deal with resource if the operation succeeded.
+    if backup.exists():
+        print("Backup was created before the cancel completed.")
+        backup.delete()
+        print("Backup deleted.")
+    else:
+        print("Backup creation was successfully cancelled.")
+# [END spanner_cancel_backup]
+
+
+# [START spanner_list_backup_operations]
+def list_backup_operations(instance_id, database_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the CreateBackup operations.
+    filter_ = (
+        "(metadata.database:{}) AND "
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.CreateBackupMetadata)"
+    ).format(database_id)
+    operations = instance.list_backup_operations(filter_=filter_)
+    for op in operations:
+        metadata = op.metadata
+        # List the pending backups on the instance.
+        if metadata.progress.progress_percent < 100:
```

Comment:
I realized that this actually can't be tested in the samples because we always wait for backups to complete. So we should remove this check and say "List all create backup operations on this database."

That also reminded me that there's no test for this below. Is that intentional?